### PR TITLE
Adds bits to set domain and enable tlse for adoption multinode ci jobs

### PR DESCRIPTION
**Pull request by marios**
As part of [1] this aims to enable tls for the adoption multinode ci. We
have to move some of the overcloud config to j2 templates so we can
include the right values for cloud_domain (dns domain). It also adds
the external network for the tripleo controllers as without it the tls
deployment code fails. Finally this includes the startup of the freeipa
server as a container on the undercloud. Also see related patch at [2].

[1] https://issues.redhat.com/browse/OSPRH-8973
[2] https://review.rdoproject.org/r/c/rdo-jobs/+/54102

### CODE REVIEW

File: devsetup/tripleo/tripleo_install.sh [hunk 1]
```diff
@@ -169,6 +169,57 @@ if [ "$EDPM_COMPUTE_CEPH_ENABLED" = "true" ] ; then
...
+export CLOUD_DOMAIN=$CLOUD_DOMAIN
+export UNDERCLOUD_FQDN=undercloud.$CLOUD_DOMAIN
+export IPA_DOMAIN=$CLOUD_DOMAIN
+export IPA_REALM=$(echo $IPA_DOMAIN | awk '{print toupper($0)}')
+export IPA_HOST=ipa.$IPA_DOMAIN
+export IPA_SERVER_HOSTNAME=$IPA_HOST
+ansible-playbook /usr/share/ansible/tripleo-playbooks/undercloud-ipa-install.yaml
+EOF
+    chmod 754 /home/zuul/ipaserver_install.sh
+    sudo /home/zuul/ipaserver_install.sh
```
Comment (marios) [inline devsetup/tripleo/tripleo_install.sh:220]: this part can be kept as non sudo since it completes ok without.

File: devsetup/tripleo/tripleo_install.sh [hunk 2]
```diff
@@ -169,6 +169,57 @@ if [ "$EDPM_COMPUTE_CEPH_ENABLED" = "true" ] ; then
...
+        -e IPA_SERVER_IP=10.255.255.25 \
+        -e PASSWORD=$IPA_ADMIN_PASSWORD \
+        -h $IPA_SERVER_HOSTNAME \
+        --read-only --tmpfs /run --tmpfs /tmp \
+        -v /sys/fs/cgroup:/sys/fs/cgroup:ro \
+        -v /tmp/ipa-data:/data:Z quay.io/freeipa/freeipa-server:fedora-39 no-exit \
+        -U -r $IPA_REALM --setup-dns --no-reverse --no-ntp \
+        --no-dnssec-validation --auto-forwarders
+    timeout 900s grep -qEi '(INFO The ipa-server-install command was successful|ERROR The ipa-server-install command failed)' <(sudo tail -F /tmp/ipa-data/var/log/ipaserver-install.log)
+    cat  <<EOF > ipa_resolv.conf
```
Comment (marios) [inline devsetup/tripleo/tripleo_install.sh:201]: need this on all the overcloud nodes too

**Comment by marios** [comment]
setting ready for review. With the latest changes we got a green run at https://softwarefactory-project.io/zuul/t/rdoproject.org/build/61b17dc1e88f4461a54898d451e42d46

(the subsequent run failed but i think it was a transient issue/re-running)

File: devsetup/tripleo/tripleo_install.sh (hunk 2 above)
Comment (cescgina) [reply devsetup/tripleo/tripleo_install.sh:204]: @marios I'm thinking that it might be a good idea to leave a note here, that this code assumes that the overcloud nodes are setup to have the ipa server as dns resolver. We do that in CI from a rdo-jobs playbook, but it's not obvious from inspection of the code here

File: devsetup/tripleo/undercloud.conf.j2 [hunk 3]
```diff
@@ -30,13 +30,13 @@ local_mtu = {{ interface_mtu }}
 # Undercloud services. Only used with SSL. (string value)
 # Deprecated group/name - [DEFAULT]/undercloud_public_vip
 #undercloud_public_host = 192.168.24.2
-undercloud_public_host = 192.168.122.122
+undercloud_public_host = 192.168.122.99
```
Comment (cescgina) [inline devsetup/tripleo/undercloud.conf.j2:33]: what is the reason for this change? was this ip unused until now, so we had a placeholder value?

Comment (cescgina) [reply devsetup/tripleo/undercloud.conf.j2:33]: ack, thanks I assumed that was the case but wanted to clarify

File: devsetup/tripleo/tripleo_install.sh (hunk 2 above)
Comment (marios) [reply devsetup/tripleo/tripleo_install.sh:204]: ack

**Comment by marios** [comment]
holding for green run on the ceph/non tls job

**Comment by marios** [comment]
> holding for green run on the ceph/non tls job

https://review.rdoproject.org/r/c/testproject/+/54103/3#message-94c5659e701ae38723659a9b2646be2e2f44720e
